### ISSUE TITLE
Best way for early stopping of the code.

### ISSUE THREAD

**Issue by imsrgadich** [GPflow/GPflow#876]
Hi devs, I'm running my deep Gaussian processes code with GPFlow. In some of the simulations, my code was over-optimizing. There is a fear of over-fitting in those cases. What is the best way to avoid that?

Thank you.

**Comment by hughsalimbeni** [comment]
I see two ways of interpreting your question:
* what are good criteria for early stopping in deep GP models? I certainly don't know the answer to this in general (but the Bayesian in me says that this shouldn't be necessary...). I suppose a test likelihood on a validation set would be one way to go. 
* how to implement early stopping with a gpflow optimizer? This is easy if you set up the optimization loop yourself. There is an example of this in #820. You'd need to implement the stopping criterion yourself (say through (the gradient of) a validation loss)

**Comment by st--** [comment]
@imsrgadich this is the kind of question better asked on stackoverflow - there's a gpflow tag now! https://stackoverflow.com/questions/tagged/gpflow
